src/phoenix/experimental/evals/functions/binary.py

```diff
@@ -58,4 +51,80 @@ def llm_eval_binary(
         )
 
     responses = model.generate(prompts.to_list(), system_instruction)
-    return responses
+    rail_classes = set(rails)
+    return [
+        (rail_class if (rail_class := resp.strip()) in rail_classes else None) for resp in responses
+    ]
```

Comment:
does this not have errors? probably need to test this a bit.

Reply:
Added unit tests in [0a55a9c](https://github.com/Arize-ai/phoenix/pull/1250/commits/0a55a9cc8c7a8804e8471e3651b61127bb944924) that should cover the conditions in this line.